### PR TITLE
New Script: NextPVR

### LINKED ISSUE #384: NextPVR
https://github.com/community-scripts/ProxmoxVE/discussions/164

Application Name
NextPVR

Website
https://www.nextpvr.com/

Description
NextPVR is a personal video recorder application for Microsoft Windows, Linux, Mac and Docker. NextPVR makes it easy to watch or record live TV, and provides great features like series recordings, web scheduling, iPhone/iPad client application, Kodi/Emby integration etc.

Wiki: https://github.com/sub3/NextPVR/wiki/

### PR DESCRIPTION
**Pull request by MickLesk**
## Description

Add new LXC: NextPVR. 
I only can test the usage of the ui. i dont have an System for it. 
After Merge we can add the requester as final reviewer / or need later an bugfix.

Fixes # (issue)
Fixes https://github.com/community-scripts/ProxmoxVE/issues/384

## Type of change

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [x] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

### CODE REVIEW

**Review by havardthom — DISMISSED** [review]

File: install/nextpvr-install.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies (Patience)"
+$STD apt-get install -y \
```
[the remark this answers is not in the document]
Comment (MickLesk) [reply install/nextpvr-install.sh:17]: Added little by little until dpkg worked, the manual / documentation was rather bad than good.

File: json/nextpvr.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "NextPVR",
+  "slug": "nextpvr",
+  "categories": [
+    0
```
[the remark this answers is not in the document]
Comment (MickLesk) [reply json/nextpvr.json:5]: Can be, but not really Sure 😅

[the remark this answers is not in the document]
Comment (havardthom) [reply json/nextpvr.json:5]: google says `NextPVR is a whole-home digital video recorder` aka DVR

File: ct/nextpvr.sh [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+msg_info "Stopping ${APP}"
+systemctl stop nextpvr-server
+msg_ok "Stopped ${APP}"
+
+msg_info "Updating ${APP}"
+apt-get update &>/dev/null
+apt-get -y upgrade &>/dev/null
+cd /opt
+wget -q https://nextpvr.com/nextpvr-helper.deb
+DEBIAN_FRONTEND=noninteractive dpkg -i nextpvr-helper.deb &>/dev/null
```
[the remark this answers is not in the document]
Comment (MickLesk) [reply ct/nextpvr.sh:70]: I'll check it again tomorrow on the PC, there was a reason why it was in there 🤔 maybe I was confused after 10 hours on the PC 😅

File: ct/nextpvr.sh [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+header_info
+check_container_storage
+check_container_resources
+if [[ ! -d /opt/nextpvr ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+msg_info "Stopping ${APP}"
+systemctl stop nextpvr-server
+msg_ok "Stopped ${APP}"
+
+msg_info "Updating ${APP}"
+apt-get update &>/dev/null
```
[the remark this answers is not in the document]
Comment (havardthom) [reply ct/nextpvr.sh:66]: Alright, maybe we should have it in its own section, and we can start adding it to all update functions?

```
msg_info "Updating LXC packages"
apt-get update &>/dev/null
apt-get -y upgrade &>/dev/null
msg_ok "Updated LXC packages"
```

**Comment by MickLesk** [comment]
@havardthom Done & Tested | Approve pending

File: json/nextpvr.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  "documentation": "https://github.com/sub3/NextPVR/wiki",
+  "website": "https://nextpvr.com/",
+  "logo": null,
+  "description": "NextPVR is a personal video recorder application for Microsoft Windows, Linux, Mac and Docker. NextPVR makes it easy to watch or record live TV, and provides great features like series recordings, web scheduling, iPhone/iPad client application, Kodi/Emby integration etc.",
+  "install_methods": [
+    {
+      "type": "default",
+      "script": "/ct/nextpvr.sh",
+      "resources": {
+        "cpu": 1,
```
Comment (havardthom) [inline json/nextpvr.json:21]:
```suggestion
        "cpu": "1",
```

File: json/nextpvr.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        "ram": 1024,
```
Comment (havardthom) [inline json/nextpvr.json:22]:
```suggestion
        "ram": "1024",
```

File: json/nextpvr.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        "hdd": 5,
```
Comment (havardthom) [inline json/nextpvr.json:23]:
```suggestion
        "hdd": "5",
```

**Review by havardthom — DISMISSED** [review]
These are strings in all other json's. Perhaps they should be numbers, but then it should be fixed everywhere

**Comment by MickLesk** [comment]
> These are strings in all other json's. Perhaps they should be numbers, but then it should be fixed everywhere

I was wondering, because it doesn't actually need “[INT]” if it's just numbers? But okay, it's adjusted ^^

**Review by havardthom — APPROVED** [review]

File: install/nextpvr-install.sh [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies (Patience)"
```
Comment (judeibe) [inline install/nextpvr-install.sh:16]: Is this need `(Patience)` other scripts do not have this added.

File: install/nextpvr-install.sh [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  acl \
+  dvb-tools \
+  libdvbv5-0 \
+  dtv-scan-tables \
+  libc6-dev \
+  ffmpeg \
+  mc
+msg_ok "Installed Dependencies"
+
+msg_info "Setup NextPVR (Patience)"
```
Comment (judeibe) [inline install/nextpvr-install.sh:33]: See my other comment

**Review by newzealandpaul — APPROVED** [review]

**Pull request merged**